modules/git/blob.go

```diff
@@ -50,6 +51,27 @@ func (b *Blob) GetBlobContent() (string, error) {
 	return string(buf), nil
 }
 
+// GetBlobLineCount gets line count of lob as raw text
+func (b *Blob) GetBlobLineCount() (int, error) {
+	reader, err := b.DataAsync()
+	if err != nil {
+		return 0, err
+	}
+	buf := make([]byte, 32*1024)
+	count := 0
+	lineSep := []byte{'\n'}
```

Comment:
It seems to be that git blob object will convert line ending to `\n`. Since we get the reader from `EncodedObject`, it's probably okay to count `\n`.  🤔

Reply:
It doesn't work for Mac (not that I have one). I've pushed the same file with 7 lines: LF, CRLF and CR. The first two reported 7 lines; the last one reported only one.

Reply:
CRLF works fine because LF is in it.

Reply:
~~My suggestion is to count both `\n` and `\r` and take the larger number.~~

Oh, but! The diff tool doesn't recognize `\r` either! 😅  So having the correct line count is pointless. That should be a different PR, then.